platforms/software/testing-base/src/main/java/org/gradle/api/tasks/testing/AbstractTestTask.java

```diff
@@ -545,12 +544,12 @@ private void handleCollectedResults(TestCountLogger testCountLogger) {
             handleTestFailures();
         } else if (testCountLogger.getTotalTests() == 0) {
             // No tests were executed, the following rules apply:
-            // - If there are no filters, and no tests or test suites were discovered, emit a deprecation warning
+            // - If there are no filters, and no tests or test suites were discovered, fail
             // - If there are filters and the task is configured to fail when no tests match the filters, throw an exception
             // - Otherwise, this is fine - the task should succeed with no warnings or errors
             if (testsAreNotFiltered()) {
                 if (testCountLogger.getTotalDiscoveredItems() == 0) {
-                    emitDeprecationMessage();
+                    throw new TestExecutionException("There are test sources present and no filters are applied, but the test task did not discover any tests to execute. This is likely due to a misconfiguration.  Please check your test configuration.");
```

Comment:
This gained an extra space:
```suggestion
                    throw new TestExecutionException("There are test sources present and no filters are applied, but the test task did not discover any tests to execute. This is likely due to a misconfiguration. Please check your test configuration.");
```